components/google_sheets/sources/new-spreadsheet/new-spreadsheet.js

```diff
@@ -0,0 +1,95 @@
+const common = require("../common");
+
+module.exports = {
+  ...common,
+  key: "google_sheets-new-spreadsheet",
+  name: "New Spreadsheet (Instant)",
+  description:
+    "Emits an event each time a new spreadsheet is created in a drive.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    folders: {
+      propDefinition: [
+        common.props.google_sheets,
+        "folders",
+        (c) => ({ driveId: c.watchedDrive }),
+      ],
+    },
+  },
+  hooks: {
+    ...common.hooks,
+    async deploy() {
+      this._setLastFileCreatedTime(Date.now());
+    },
+  },
+  methods: {
+    ...common.methods,
+    shouldProcess(file) {
+      const watchedFolders = new Set(this.folders);
+      return (
+        watchedFolders.size == 0 ||
+        file.parents.some((p) => watchedFolders.has(p))
+      );
+    },
+    async getSpreadsheetToProcess(event) {
+      const pageToken = this._getPageToken();
+      const {
+        changedFiles,
+        newStartPageToken,
+      } = await this.google_sheets.getChanges(
+        pageToken,
+        this.watchedDrive === "myDrive" ? null : this.watchedDrive
+      );
+      this._setPageToken(newStartPageToken);
+
+      const file = changedFiles
+        .filter((file) => file.mimeType.includes("spreadsheet"))
+        .shift();
+      if (file && this.shouldProcess(file)) return file;
+    },
+    async processSpreadsheet(file) {
+      const lastFileCreatedTime = this._getLastFileCreatedTime();
+      let maxCreatedTime = lastFileCreatedTime;
+
+      const fileInfo = await this.google_sheets.getFile(file.id);
+      const createdTime = Date.parse(fileInfo.createdTime);
+      if (createdTime > maxCreatedTime) maxCreatedTime = createdTime;
+      if (createdTime <= lastFileCreatedTime) return;
+
+      const meta = this.generateMeta(file, createdTime);
+      this.$emit(fileInfo, meta);
+
+      this._setLastFileCreatedTime(maxCreatedTime);
+    },
+    generateMeta(file, createdTime) {
+      return {
+        summary: `New File ID: ${file.id}`,
+        id: file.id,
+        ts: createdTime,
+      };
+    },
+    _getLastFileCreatedTime() {
+      return this.db.get("lastFileCreatedTime");
+    },
+    _setLastFileCreatedTime(lastFileCreatedTime) {
+      this.db.set("lastFileCreatedTime", lastFileCreatedTime);
+    },
+    takeSheetSnapshot() {},
```

Comment:
Now that I see this I realize that this event source is extending the Google Sheets base module, which might not be the correct approach since that module assumes that the source of events are the contents of the spreadsheets themselves. This event source on the other hand is closer to Google Drive, and you should be able to achieve the same behaviour by just extending `google_drive-new-files-instant` and overriding the `shouldProcess` method, right?

Reply:
Updated to extend google_drive-new-files-instant